### PR TITLE
Creative API: embed-context

### PR DESCRIPTION
**Pull request by nekodo**
Adds an embed-context postMessage API which provides access to some metadata about the embed (location.href, referrer, canonicalUrl, pageViewId, clientId). The API consists of send-embed-context message sent by the creative, to which AMP runtime responds with embed-context message containing the data.

This change (along with with #4021 and #4023) implements the postMessage API of #3019.

### CODE REVIEW

File: src/3p-frame.js [hunk 1]
```diff
@@ -128,10 +132,24 @@ export function getIframe(parentWindow, element, opt_type) {
...
+ * Creates the iframe for the embed. Applies correct size and passes the embed
+ * attributes to the frame via JSON inside the fragment.
+ * @param {!Window} parentWindow
+ * @param {!Element} element
+ * @param {string=} opt_type
+ * @return {!Element} The iframe.
+ */
+export function getIframe(parentWindow, element, opt_type) {
+  return getIframe(parentWindow, element, opt_type).iframe;
+>>>>>>> f4f1250b91b0cc22896d01e6c7132ba4108a1ae3
```
Comment (erwinmombay) [inline src/3p-frame.js]: remove git diff marker

Comment (nekodo) [inline src/3p-frame.js]: Yeah, this one is not quite yet ready for review:)

Comment (erwinmombay) [inline src/3p-frame.js]: haha yeah, just passing through 💨

File: src/3p-frame.js [hunk 2]
```diff
@@ -128,6 +128,7 @@ export function getIframe(parentWindow, element, opt_type) {
   };
   iframe.setAttribute(
       'data-amp-3p-sentinel', attributes._context.amp3pSentinel);
+  iframe._context = attributes._context;
```
Comment (nekodo) [inline src/3p-frame.js]: Any better idea for how to pass the context metadata out of getIframe?

**Comment by lannka** [comment]
Seems obsolete. Most of the data is available via [window.context](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-information)

**Pull request closed without merging**